scheduler/scheduler-server/src/main/java/org/ow2/proactive/utils/PCAProxyRule.java

```diff
@@ -140,12 +140,22 @@ private String rewriteTarget(String target, String endpointPath, HttpServletRequ
                 logger.debug(String.format("Rewrote %s to %s", target, newTarget));
             }
             referrerCache.put(target, endpointPath);
-            if (request.getMethod().equals(HttpMethod.GET.asString())) {
+            if (HttpMethod.GET.is(request.getMethod())) {
                 redirectGetRequest(target, endpointPath, request, response, newTarget);
             }
             return newTarget;
         } else {
             logger.trace("Target already contains endpoint");
+            if (!target.startsWith(endpointPath)) {
+                // endpoint is in the middle of the path (who did this?), let's bring it on top
```

Comment:
how do you get the endpoint ? Through the pca services ?

Reply:
In the request directly, thanks to a regular expression (see previous pull request)

Reply:
https://github.com/ow2-proactive/scheduling/commit/96ec7c0cc8dd3749937ce7df70572e453c5151ea